sys/arm64/arm64/hyp_stub.S

```diff
@@ -37,7 +37,14 @@
  */
 .macro vector_stub_el1h_sync
 	.align 7
+#if __has_feature(capabilities)
+#ifndef __CHERI_PURE_CAPABILITY__
```

Comment:
I do wonder if this interface shouldn't always be a capability?

Reply:
That does seem a bit cleaner to me. What would be the right way to create one in a hybrid kernel? Derive from `cheri_getdefault()`?

Reply:
You can use a `__cheri_tocap` pointer cast to derive one from DDC, but `cheri_getdefault` might be a smaller #ifdef.

Reply:
Well you'd want to derive it from EL2's cap, technically, no?

Reply:
(and code not data cap, if you have that distinction)

Reply:
I'm still not fully sure what the right answer is here.  As Jess noted in the other thread, this is using `cvtp`, so deriving from PCC and not DDC_EL2.  That might be the right answer for the hybrid interface.  It's actually a bit wonky even for the purecap case to pass a capability derived in EL1 for use in EL2.  Maybe best to just use what is here for now and add a suitable comment somewhere.

Reply:
It's not hard to change the hybrid kernel to pass a capability, I'm testing that change now. I can't use PCC in purecap kernels of course, so I'm instead deriving from `kernel_root_cap`. (I forgot to AND the perms of that capability with those of the PCC, so I made that change as well, will push shortly.)

There is an EL2 capability, but it seems slightly inappropriate to use that here since it's for EL2's virtual address space, whereas here we're passing a physical address. EL2 is currently identity-mapped so it's probably not an important distinction, but it does seem a bit confusing.